BrowserKit/Sources/ComponentLibrary/BottomSheet/BottomSheetViewController.swift

```diff
@@ -147,7 +149,7 @@ public class BottomSheetViewController: UIViewController,
     }
 
     public var currentWindowUUID: WindowUUID? {
-        return (self.view as? ThemeUUIDIdentifiable)?.currentWindowUUID
+        return windowUUID
```

Comment:
@DanielDervishi Isn't `BottomSheetViewController` one of the components which is potentially reused? My only concern is that I vaguely remember seeing that there were scenarios in which this VC was could be presented on more than one window during its lifetime. (I could be misremembering though, it has been a while since I looked at this specific area.)

Reply:
@mattreaganmozilla Yes, this is part of the component library, so is reusable. For our other views, we are injecting `windowUUID`, can you confirm what the best way to set the windowUUID here? It was giving a nil error causing the dimmed background code to not be triggered.

Reply:
@cyndichin So I don't have 100% context on the specifics here, but just as a general rule, injection of the UUID at the time of instantiation won't work correctly if the same VC/view is later presented on a different window obviously. So we would need to either A) inject/update the UUID at the time of presentation or B) figure out why the `theme` is being requested before the view is actually installed in a window (if it's installed in a window, the `currentWindowUUID` as written shouldn't be nil, so it sounds like we're trying to `applyTheme` perhaps before it's actually presented anywhere).

Reply:
@DanielDervishi @cyndichin Can you LMK what the repro steps are for this (the linked ticket is blank)? I can add some extra eyes if it would be helpful.

Reply:
@DanielDervishi @cyndichin Fyi I took another look at this and I can't actually see any place where we're reusing existing instances of `BottomSheetViewController` (which was my original concern mentioned above). So I think injecting the UUID in the initializer is fine.

Reply:
@mattreaganmozilla Sorry for the delay, haven't gotten to look at this. Ohh I misinterpreted the original concern. We are reusing the Bottom Sheet in multiple places. However, it seems that the concern is only if we instantiate a single Bottom Sheet and use the same for multiple windows, then that can be the issue. As opposed to a new one being instantiated for each window. 

Can you point to an area in which this issue occurs and we aren't injecting a windowUUID to gain more clarity?

Reply:
Adding some context because I had a similar issue. It seems that windowUUID is taken from the window property of the view. The window seems to be nil when the view controller is presented modally and is only available after viewDidAppear. This makes applying theming and adding colors a bit glitchy since the view controller is already on screen.